platform/pulp/tasking/tasks.py

```diff
@@ -200,15 +201,26 @@ def apply_async_with_reservation(self, resource_type, resource_id, *args, **kwar
         with transaction.atomic():
             task_status = TaskStatus.objects.create(pk=inner_task_id, state=TaskStatus.WAITING,
                                                     group=group_id, **parent_arg)
-            for tag in tag_list:
+            for tag in tags:
                 task_status.tags.create(name=tag)
 
+        # Wrap celery signature into kwargs for _queue_reserved_task
+        async_args = tuple()
+        async_kwargs = {
+            "tags": tags,
```

Comment:
This wasn't exactly part of the story, but I don't think that tags and group_id should be kept in the async_kwargs. They are only needed in the few statements above and do not need to be passed along to the task.